mkosi/config.py

```diff
@@ -5244,6 +5245,35 @@ def finalize_term() -> str:
     return term if sys.stderr.isatty() else "dumb"
 
 
+def finalize_git_config(proxy_url: Optional[str], env: dict[str, str]) -> dict[str, str]:
+    if proxy_url is None:
+        return {}
+
+    cnt_str = env.get("GIT_CONFIG_COUNT", "0")
+    if not cnt_str.isdigit():
+        raise ValueError("GIT_CONFIG_COUNT environment variable must be set to a valid integer")
+    cnt = int(cnt_str)
+
+    env_override = {}
+    for i in range(cnt):
+        key = env.get(f"GIT_CONFIG_KEY_{i}")
+        if key is None:
+            raise ValueError(
+                f"GIT_CONFIG_KEY_{i} environment variable must be set with GIT_CONFIG_COUNT={cnt}"
+            )
+        if key in ("http.proxy", "https.proxy"):
+            env_override[f"GIT_CONFIG_VALUE_{i}"] = proxy_url
+
+    # Override HTTP/HTTPS proxy in case its set in .gitconfig to a different value than proxy_url.
+    return env_override | {
+        "GIT_CONFIG_COUNT": str(cnt + 2),
+        f"GIT_CONFIG_KEY_{cnt}": "http.proxy",
+        f"GIT_CONFIG_VALUE_{cnt}": proxy_url,
+        f"GIT_CONFIG_KEY_{cnt + 1}": "https.proxy",
+        f"GIT_CONFIG_VALUE_{cnt + 1}": proxy_url,
+    }
```

Comment:
Is git first setting wins or last setting wins? If it's the latter, wouldn't just setting these last be enough?

Reply:
Appears to be last setting wins from my local testing and docs: https://git-scm.com/docs/git-config#Documentation/git-config.txt-get

So my previous nonsense from before is indeed unnecessary!

Reply:
Fixed in newest version